src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -1616,63 +1626,93 @@ private function integerRangeMath(Type $range, BinaryOp $node, Type $operand): T
 				}
 			}
 		} elseif ($node instanceof Expr\BinaryOp\Mul) {
-			if ($operand instanceof ConstantIntegerType) {
-				/** @var int|float|null $min */
-				$min = $rangeMin !== null ? $rangeMin * $operand->getValue() : null;
+			$min1 = ($rangeMin ?? -INF) * ($operandMin ?? -INF);
+			$min2 = ($rangeMin ?? -INF) * ($operandMax ?? INF);
+			$max1 = ($rangeMax ?? INF) * ($operandMin ?? -INF);
+			$max2 = ($rangeMax ?? INF) * ($operandMax ?? INF);
 
-				/** @var int|float|null $max */
-				$max = $rangeMax !== null ? $rangeMax * $operand->getValue() : null;
-			} else {
-				/** @var int|float|null $min */
-				$min = $rangeMin !== null && $operand->getMin() !== null ? $rangeMin * $operand->getMin() : null;
+			$min = min($min1, $min2, $max1, $max2);
+			$max = max($min1, $min2, $max1, $max2);
 
-				/** @var int|float|null $max */
-				$max = $rangeMax !== null && $operand->getMax() !== null ? $rangeMax * $operand->getMax() : null;
+			if ($min === -INF) {
+				$min = null;
 			}
-
-			if ($min !== null && $max !== null && $min > $max) {
-				[$min, $max] = [$max, $min];
-			}
-
-			// invert maximas on multiplication with negative constants
-			if ((($range instanceof ConstantIntegerType && $range->getValue() < 0)
-					|| ($operand instanceof ConstantIntegerType && $operand->getValue() < 0))
-				&& ($min === null || $max === null)) {
-				[$min, $max] = [$max, $min];
+			if ($max === INF) {
+				$max = null;
 			}
-
 		} else {
 			if ($operand instanceof ConstantIntegerType) {
 				$min = $rangeMin !== null && $operand->getValue() !== 0 ? $rangeMin / $operand->getValue() : null;
 				$max = $rangeMax !== null && $operand->getValue() !== 0 ? $rangeMax / $operand->getValue() : null;
 			} else {
-				$min = $rangeMin !== null && $operand->getMin() !== null && $operand->getMin() !== 0 ? $rangeMin / $operand->getMin() : null;
-				$max = $rangeMax !== null && $operand->getMax() !== null && $operand->getMax() !== 0 ? $rangeMax / $operand->getMax() : null;
-			}
+				// Avoid division by zero when looking for the min and the max by using the closest int
+				$operandMin = $operandMin !== 0 ? $operandMin : 1;
+				$operandMax = $operandMax !== 0 ? $operandMax : -1;
+
+				if (
```

Comment:
I found really complicated to compute
```
int<-1, 2> / int<-1, 2>
```
It seemed easier to split negative and positive integers.

So
```
int<-1, 0> / int<-1, 0>
int<-1, 0> / int<0, 2>
int<0, 2> / int<-1, 0>
int<0, 2> / int<0, 2>
```
And then I merge all these types.